Makefile

```
@@ -24,7 +25,18 @@ else
 	STRIP_FLAG := -strip=$(STRIP_PATH)
 endif
 
-GOARCH ?= $(shell go env GOARCH)
+ifndef CGO_ENABLED
+CGO_ENABLED_NDEF := 1
+CGO_ENABLED := $(shell go env CGO_ENABLED)
+endif
+ifndef CC
+CC_NDEF := 1
+CC := $(shell go env CC)
+endif
+ifndef GOARCH
+GOARCH_NDEF := 1
+GOARCH := $(shell go env GOARCH)
+endif
```

`?=` can be used to define a variable if it hasn't been set.

```suggestion
CGO_ENABLED ?= $(shell go env CGO_ENABLED)
CC ?= $(shell go env CC)
GOARCH ?= $(shell go env GOARCH)
```

---

这里是需要覆写哈，不是这个逻辑